### DISCUSSION TITLE
Planned workflow termination

### DISCUSSION THREAD

**Discussion by Fraer** [argoproj/argo-workflows#6935] [Q&A] [answered]
Hi, does argo allows to terminate a workflow at a specific moment in time (if it is still running of course), kind of a termination walltime or max allowed duration?
My use case is pretty simple, i'm running some migration scripts as workflow step that needs to be stopped before the beginning of a working day.

**Comment by terrytangyuan** [comment, marked as answer]
You can specify activeDeadlineSeconds in your workflow spec: https://argoproj.github.io/argo-workflows/fields/#workflowspec
